console/src/components/MapCore/MapCore.js

```diff
@@ -560,16 +581,55 @@ class MapCoreBase extends React.Component {
 		delete rest.proposedDestination;
 		delete rest.setDestination;
 		delete rest.skin;
+		delete rest.topLocations;
 		delete rest.updateNavigation;
 		delete rest.viewLockoutDuration;
 		delete rest.zoomToSpeedScaleFactor;
+		const {duration, distance, eta, selectedDestination, selfDriving} = this.state;
+
 		return (
 			<div {...rest} className={classnames(className, css.map)}>
 				{this.message ? <div className={css.message}>{this.message}</div> : null}
-				<nav className={css.tools}>
-					{tools}
-					<ToggleButton alt="Follow" selected={this.state.follow} underline icon="forward" onClick={this.changeFollow} />
-				</nav>
+				<Column className={css.tools} align="stretch space-between">
+					{
+						selfDrivingSelection && <div>
+							<Divider>SELF DRIVING</Divider>
+							<Button small>AUTO</Button>
+							<Button small>MANUAL</Button>
+						</div>
+					}
+					<div>
+						<Divider>TOP LOCATIONS</Divider>
+						<Group childComponent={Button} onSelect={this.estimateRoute} selectedProp="selected">
+							{this.topLocations ? this.topLocations.map(({properties}) => {
+								const {index, description} = properties;
+								return {
+									children: `${index} - ${description}`,
+									className: css.button,
+									key: `${description}-btn`,
```

Comment:
Let's also append the `index` value to this key in case the descriptions happen to be the same.